### PR TITLE
Sector Z, Fortuna & General Docs

### PR DESCRIPTION
**Pull request by sonicdcer**

### CODE REVIEW

File: src/overlays/ovl_i4/fox_sz.c [hunk 1]
```diff
@@ -120,132 +129,142 @@ void SectorZ_Missile_Update(Actor* this) {
...
-        if (sp24) {}                                    //! FAKE
-        sp2C = fabsf(this->fwork[4] - this->obj.pos.x); //! FAKE
-        sp20 = SIN_DEG((this->index * 45) + gGameFrameCount) * 5000.0f;
-        sp1C = COS_DEG((this->index * 45) + (gGameFrameCount * 2)) * 5000.0f;
+        if (z) {} //! FAKE
+        z = fabsf(this->fwork[6] - this->obj.pos.z); //! FAKE
+
+        // Escorts wooble movement
+        xPitch = SIN_DEG((this->index * 45) + gGameFrameCount) * 5000.0f;
+        yPitch = COS_DEG((this->index * 45) + (gGameFrameCount * 2)) * 5000.0f;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:148]: This is them flying around the missile during the first missile cutscene.

File: src/overlays/ovl_i4/fox_sz.c [hunk 2]
```diff
@@ -120,132 +129,142 @@ void SectorZ_Missile_Update(Actor* this) {
...
+        // For missile
+        xPitch = yPitch = 0.0f;
     }
 
-    this->fwork[4] = gBosses->obj.pos.x + sp20 + 400.0f;
-    this->fwork[5] = gBosses->obj.pos.y + sp1C + 100.0f;
-    this->fwork[6] = gBosses->obj.pos.z;
+    this->fwork[MISSILE_TARGET_X] = gBosses[0].obj.pos.x + xPitch + 400.0f;
+    this->fwork[MISSILE_TARGET_Y] = gBosses[0].obj.pos.y + yPitch + 100.0f;
+    this->fwork[MISSILE_TARGET_Z] = gBosses[0].obj.pos.z;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:156]: fwork 4-6 for all range actors appears to be a general purpose target position, so I don't think we need this to be missile-specific.

File: src/overlays/ovl_i4/fox_sz.c [hunk 3]
```diff
@@ -120,132 +129,142 @@ void SectorZ_Missile_Update(Actor* this) {
...
-    actor->obj.id = OBJ_ACTOR_ALLRANGE;
-    actor->aiType = arg1 + AI360_10 + 3;
+// Enemies that follow the missile
+void SectorZ_MissileEscort(ActorAllRange* this, s32 enemyIndex) {
+    Actor_Initialize(this);
+    this->obj.status = OBJ_INIT;
+    this->obj.id = OBJ_ACTOR_ALLRANGE;
+    this->aiType = enemyIndex + AI360_10 + 3;
 
-    actor->obj.pos.x = gActors[10].obj.pos.x + D_i4_8019F4E4[arg1].x;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:202]: For this level, defines for the missile actor slots seems like a good idea. we could then write AI360_10 + 3 in terms of those.

File: src/overlays/ovl_i4/fox_sz.c [hunk 4]
```diff
@@ -505,21 +537,22 @@ void SectorZ_UpdateEvents(Actor* actor) {
             break;
 
         case 3:
-            actor8 = &gActors[8];
+            katt = &gActors[8];
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:540]:
```suggestion
            katt = &gActors[AI360_KATT];
```

File: src/overlays/ovl_i4/fox_sz.c [hunk 5]
```diff
@@ -620,15 +653,15 @@ void SectorZ_UpdateEvents(Actor* actor) {
...
+                this->timer_0BE = 550;
             }
             break;
 
         case 100000:
             break;
     }
 
-    if ((gGameFrameCount & 0x18) == 0) {
+    if ((gGameFrameCount & 24) == 0) {
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:664]: I prefer this one in hex since it's a bitwise mask.

File: src/overlays/ovl_i4/fox_sz.c [hunk 6]
```diff
@@ -639,129 +672,136 @@ void SectorZ_UpdateEvents(Actor* actor) {
...
+Vec3f sTeamSetupOffsetPos[] = {
     { 0.0f, 20000.0f, 0.0f }, { -1000.0f, 100.0f, 500.0f }, { -1300.0f, -50.0f, 0.0f }, { -1600.0f, 50.0f, -500.0f }
 };
 
-void SectorZ_8019B888(void) {
-    s32 i;
-    Actor* actor;
+void SectorZ_TeamSetup(void) {
+    TeamId i;
+    ActorAllRange* SFTeam;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:778]: I've just been using `team`

File: src/overlays/ovl_i4/fox_sz.c [hunk 7]
```diff
@@ -1713,81 +1756,84 @@ void SectorZ_Boss_Update(Boss* boss) {
...
 }
 
-void SectorZ_Boss_Draw(Boss* boss) {
+void SectorZ_GreatFoxDraw(GreatFoxSZ* this) {
     gSPFogPosition(gMasterDisp++, gFogNear, 1005);
     Cutscene_DrawGreatFox();
 }
 
-void SectorZ_8019EA68(void) {
+void SectorZ_SetLevelObjects(void) {
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:1831]: I'd go with `LoadLevelObjects`, as this function is analogous to `Object_LoadLevelObjects`.

File: src/overlays/ovl_i4/fox_sz.c [hunk 8]
```diff
@@ -1713,81 +1756,84 @@ void SectorZ_Boss_Update(Boss* boss) {
...
     s32 i;
     s32 j;
     Actor* actor;
     Scenery360* scenery360;
-    Boss* boss = &gBosses[0];
+    GreatFoxSZ* boss = &gBosses[0];
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:1836]: `greatFox` over `boss`?

File: src/overlays/ovl_i4/fox_sz.c [hunk 9]
```diff
@@ -120,132 +129,142 @@ void SectorZ_Missile_Update(Actor* this) {
...
+    AUDIO_PLAY_SFX(NA_SE_EN_PUNCH_ENGINE, this->sfxSource, 4);
 }
 
-void SectorZ_8019A0F8(Actor* actor, s32 arg1) {
-    Actor_Initialize(actor);
-    actor->obj.status = OBJ_INIT;
-    actor->obj.id = OBJ_ACTOR_ALLRANGE;
-    actor->aiType = arg1 + AI360_10 + 3;
+// Enemies that follow the missile
+void SectorZ_MissileEscort(ActorAllRange* this, s32 enemyIndex) {
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:213]: SpawnMissileEscort, I think

File: src/overlays/ovl_i4/fox_sz.c [hunk 10]
```diff
@@ -120,132 +129,142 @@ void SectorZ_Missile_Update(Actor* this) {
...
 
-    actor->fwork[1] = gActors[10].fwork[1];
-    actor->iwork[11] = 1;
+    this->fwork[1] = gActors[10].fwork[1];
+    this->iwork[11] = 1;
 }
 
-void SectorZ_8019A1D0(void) {
-    Actor* actor = &gActors[8];
+void SectorZ_Katt_Init(void) {
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_sz.c:232]: at some point we should come up with a consistent scheme for Init, Setup, Spawn, etc.

File: src/overlays/ovl_i4/fox_sz.c (hunk 1 above)
Comment (sonicdcer) [reply src/overlays/ovl_i4/fox_sz.c:148]: Yes it is. Are you suggesting there's something innacurate in the comment?

**Pull request merged**